brayns/engine/geometry/Geometry.h

```diff
@@ -0,0 +1,209 @@
+/* Copyright (c) 2015-2022, EPFL/Blue Brain Project
+ * All rights reserved. Do not distribute without permission.
+ * Responsible Author: Nadir Roman Guerrero <nadir.romanguerrero@epfl.ch>
+ *
+ * This file is part of Brayns <https://github.com/BlueBrain/Brayns>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#pragma once
+
+#include <brayns/common/Bounds.h>
+
+#include <ospray/ospray_cpp/Geometry.h>
+
+#include <cassert>
+#include <numeric>
+#include <string_view>
+#include <vector>
+
+namespace brayns
+{
+/**
+ * @brief The RenderableBoundsUpdater is a templated class that can be specialized to update bounds
+ * based on a trasnformation and a given specialized geometry
+ */
+template<typename T>
+class GeometryBoundsUpdater
+{
+public:
+    /**
+     * @brief Updates a Bounds object with the given geometry trasnformed by the given matrix
+     * @param renderableType The source renderable used to update the bounds
+     * @param transform The matrix by which to trasnform the geometry
+     * @param bounds The bounds to update
+     */
+    static void update(const T &renderableType, const Matrix4f &transform, Bounds &bounds)
+    {
+        (void)renderableType;
+        (void)transform;
+        (void)bounds;
+    }
+};
+
+/**
+ * @brief Utility class to deduce the Ospray geometry name for a given geometry type
+ */
+template<typename T>
+class OsprayGeometryName
+{
+public:
+    /**
+     * @brief Returns the Ospray name of the type
+     */
+    static const std::string &get()
```

Comment:
Usually I prefer a static inline const std:string public member that I name "of" so you avoid the parenthesis but it is fine also like that...
`OsprayGeometryName::of<T>`

Reply:
Before it was returning a string_view, so I just updated the method. Can be changed in a future iteration.